### ISSUE TITLE
Homoscedastic Loss Function

### ISSUE THREAD

**Issue by nivesh48** [Hui-Li/multi-task-learning-example-PyTorch#2]
This isn't an issue but a doubt i would like to clarify. When i am using the homoscedastic loss for my area of research the loss values are in negatives and starting to converge in negative. Is this behavior natural for this multi task loss or am i doing any mistake?

**Comment by zhackzey** [comment]
> This isn't an issue but a doubt i would like to clarify. When i am using the homoscedastic loss for my area of research the loss values are in negatives and starting to converge in negative. Is this behavior natural for this multi task loss or am i doing any mistake?

Same question. I also observed that the log_var values continue decreasing in negative, and the total loss continues decreasing. Have no idea and doubt if the model can converge.

**Comment by dariocazzani** [comment]
I think the problem arises from the fact that the implementation doesn't use ![](https://latex.codecogs.com/gif.latex?{\color{Yellow}&space;\sigma&space;^{2}})
See [Issue 3](https://github.com/Hui-Li/multi-task-learning-example-PyTorch/issues/3)

**Comment by zhackzey** [comment]
> I think the problem arises from the fact that the implementation doesn't use ![](https://camo.githubusercontent.com/58119aed4977dd0e0fb0ebeecee26a5ce4ce5c9681628e6b30aa16b5659a9959/68747470733a2f2f6c617465782e636f6465636f67732e636f6d2f6769662e6c617465783f253742253543636f6c6f7225374259656c6c6f772537442673706163653b2535437369676d612673706163653b25354525374232253744253744)
> See [Issue 3](https://github.com/Hui-Li/multi-task-learning-example-PyTorch/issues/3)
I dont think the formula is wrong. The uncertainty parameter is log \sigma^2. Thus，exp（-log\sigma^2）is 1/sigma^2.

**Comment by tong-zeng** [comment]
Hi @zhackzey I think as [Issue #3](https://github.com/Hui-Li/multi-task-learning-example-PyTorch/issues/3) reported,
The uncertainty parameter is  `σ^-2`, and the uncertainty penalty is `logσ`, when taking the exp in the code, `exp(-logσ) = σ^-1` which is different with uncertainty parameter `σ^-2` in the paper.

**Comment by tong-zeng** [comment]
Hi @zhackzey Just realized that the author corrected the formula in a new version of the paper, https://arxiv.org/pdf/1703.04977.pdf

**Comment by baojunqi** [comment]
> Hi @zhackzey Just realized that the author corrected the formula in a new version of the paper, https://arxiv.org/pdf/1703.04977.pdf

How about the performance using this loss function in your research area? I also found that my loss values are negative and I wonder if that is correct? I am really confused...
